internal/controlplane/handlers_oauth_test.go

```diff
@@ -572,7 +593,9 @@ func TestProviderCallback(t *testing.T) {
 			}
 			tc.storeMockSetup(store)
 
-			s, _ := newDefaultServer(t, store, clientFactory)
+			mockGhProviderService := mockprovsvc.NewMockGitHubProviderService(ctrl)
+			tc.buildStubs(mockGhProviderService)
+			s, _ := newDefaultServer(t, store, clientFactory, mockGhProviderService)
```

Comment:
Instead of changing the signature of `newDefaultServer` let's set `s.ghProviders = mockGhProviderService` after creating the server. That's what we do for `providerAuthManager`, `sessionService` etc.

Reply:
I do see this pattern in other places too after your comment. I tried not to change the server behavior outside of server_test but then we create a new instance so it shouldnt matter. Thanks for pointing it out 👍